server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -627,9 +627,12 @@ public VirtConsumerMap getHostConsumersMap(Owner owner, Iterable<String> hypervi
             .createAlias("owner", "o")
             .createAlias("hypervisorId", "hvsr")
             .add(Restrictions.eq("o.key", ownerKey))
-            .add(this.getHypervisorIdRestriction(hypervisorIds));
+            .add(this.getHypervisorIdRestriction(hypervisorIds))
+                .addOrder(Order.asc("hvsr.hypervisorId"));
 
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+        CandlepinQuery<Consumer> query = this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+        query.setLockMode(LockModeType.PESSIMISTIC_WRITE);
+        return query;
```

Comment:
You can simplify this a bit by using the chainable nature of the mutators in CandlepinQuery:

```
return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria)
    .setLockMode(LockModeType.PESSIMISTIC_WRITE);
```